src/com/dynatrace/loadrunner/CommandLineParser.java

```diff
@@ -0,0 +1,169 @@
+package com.dynatrace.loadrunner;
+
+import java.util.HashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommandLineParser {
+
+	public HashMap<String, String> arguments;
+
+	public CommandLineParser(String[] args) {
+
+		arguments = new HashMap<String, String>();
+
+		String key = null;
+		String value = null;
+		int index = 0;
+
+		for (int i = 0; i < args.length; i++) {
```

Comment:
logic here is rather complex, could you please extract specific cases to methods (or just add some comments)?